src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -581,8 +592,17 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 
             //if the id is not a guid, makes a guid based on the id of the model
             Guid connectorId = GuidUtility.tryParseOrCreateGuid(obj["Id"].Value<string>());
-
-            return new ConnectorModel(startPort, endPort, connectorId);
+            if(startPort != null && endPort != null)
+            {
+                return new ConnectorModel(startPort, endPort, connectorId);
+            }
+            else
```

Comment:
Just a stylistic comment; if the if ends with a return there is no need for the else. I find it clearer to just have the code after the if..return instead of having it in a bracketed else block.